scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,538 @@
+            out.append(sum(itm, np.zeros(ndim, dtype=int)))
+
+    if not out:
+        out = np.zeros((0, ndim), dtype=int)
+    else:
+        out = np.array(out)
+
+    return out
+
+
+def _build_and_solve_system(y, d, smoothing, kernel, epsilon, powers):
+    """
+    Build and solve the RBF interpolation system of equations
+
+    Parameters
+    ----------
+    y : (P, N) float ndarray
+        Data point coordinates
+    d : (P, S) float ndarray
+        Data values at `y`
+    smoothing : (P,) float ndarray
+        Smoothing parameter for each data point
+    kernel : str
+        Name of the RBF
+    epsilon : float
+        Shape parameter
+    powers : (R, N) int ndarray
+        The exponents for each monomial in the polynomial
+
+    Returns
+    -------
+    coeffs : (P + R, S) float ndarray
+    shift : (N,) float ndarray
+        Domain shift used to create the polynomial matrix
+    scale : (N,) float ndarray
+        Domain scaling used to create the polynomial matrix
+
+    """
+    lhs, rhs, shift, scale = _build_system(
+        y, d, smoothing, kernel, epsilon, powers
+        )
+    _, _, coeffs, info = dgesv(lhs, rhs, overwrite_a=True, overwrite_b=True)
+    if info < 0:
+        raise ValueError('The %d-th argument had an illegal value' % -info)
+    elif info > 0:
+        msg = 'Singular matrix'
+        nmonos = powers.shape[0]
+        if nmonos > 0:
+            pmat = _polynomial_matrix(y, powers)
+            rank = np.linalg.matrix_rank(pmat)
+            if rank < nmonos:
+                msg = (
+                    'Singular matrix. The matrix of monomials evaluated at '
+                    'the data point coordinates does not have full column '
+                    'rank (%d/%d). Consider lowering `degree` and/or setting '
+                    '`kernel` to an RBF with a lower minimum degree.' %
+                    (rank, nmonos)
+                    )
+
+        raise LinAlgError(msg)
+
+    return shift, scale, coeffs
+
+
+def _sanitize_init_args(y, d, smoothing, kernel, epsilon, degree, k):
+    """
+    Sanitize __init__ arguments for RBFInterpolator and KNearestRBFInterpolator
+    """
+    y = np.asarray(y, dtype=float, order='C')
+    if y.ndim != 2:
+        raise ValueError('`y` must be a 2-dimensional array')
+
+    ny, ndim = y.shape
+
+    dtype = complex if np.iscomplexobj(d) else float
+    d = np.asarray(d, dtype=dtype, order='C')
+    if d.shape[0] != ny:
+        raise ValueError(
+            'Expected the first axis of `d` to have length %d' % ny
+            )
+
+    if np.isscalar(smoothing):
+        smoothing = np.full(ny, smoothing, dtype=float)
+    else:
+        smoothing = np.asarray(smoothing, dtype=float, order='C')
+        if smoothing.shape != (ny,):
+            raise ValueError(
+                'Expected `smoothing` to be a scalar or have shape (%d,)' % ny
+                )
+
+    if kernel not in _AVAILABLE:
```

Comment:
I would not make this case sensitive.
```suggestion
    kernel = kernel.lower()
    if kernel not in _AVAILABLE:
```